internal/run/source.go

```diff
@@ -50,7 +50,7 @@ type SourceResult struct {
 	InputSpec    string
 	LintResult   *validation.ValidationResult
 	ChangeReport *reports.ReportResult
-	Diagnosis    *suggestions.Diagnosis
+	Diagnosis    suggestions.Diagnosis
```

Comment:
Should this be a pointer. There are a lot of nil checks

Reply:
Doesn't need to be a pointer because it's a map